itou/siaes/factories.py

```diff
@@ -66,6 +66,22 @@ class SiaeWith2MembershipsFactory(SiaeFactory):
     membership2 = factory.RelatedFactory(SiaeMembershipFactory, "siae", is_siae_admin=False)
 
 
+class SiaeWith4MembershipsFactory(SiaeFactory):
+    """
+    Generates an Siae() object with 4 members for unit tests.
+    https://factoryboy.readthedocs.io/en/latest/recipes.html#many-to-many-relation-with-a-through
+    """
+
+    # active admin user
+    membership1 = factory.RelatedFactory(SiaeMembershipFactory, "siae")
+    # active normal user
+    membership2 = factory.RelatedFactory(SiaeMembershipFactory, "siae", is_siae_admin=False)
+    # inactive admin user
+    membership3 = factory.RelatedFactory(SiaeMembershipFactory, "siae", user__is_active=False)
+    # inactive normal user
+    membership4 = factory.RelatedFactory(SiaeMembershipFactory, "siae", is_siae_admin=False, user__is_active=False)
```

Comment:
Je pinaille, mais j'aurais bien renommé les variables pour qu'elles reflètent le statut des utilisateurs (`active_admin_member`, `active_normal_member`, ...).

Reply:
J'aimerais bien aussi, mais AFAIU je dois respecter cette syntaxe de `factoryboy` pour que celui ci puisse faire sa magie. Ca sort tout droit de l'exemple sur https://factoryboy.readthedocs.io/en/latest/recipes.html#many-to-many-relation-with-a-through

Pour en avoir le coeur net je vais tout de même faire un petit test, je te dis ça asap.

Reply:
C'est KO, j'ai renommé par exemple `membership2` en `active_normal_membership2` et je me prends une :

```
factory.errors.InvalidDeclarationError: Received deep context for unknown fields: {'active_normal_membership2__user__is_active': False} (known=['address_line_1', 'auth_email', 'city', 'department', 'email', 'kind', 'naf', 'name', 'phone', 'post_code', 'siret'])
```

Reply:
re @celine-m-s